flask_server.py

```diff
@@ -110,6 +84,57 @@ def logs():
 
 ################### API ROUTES ###################
 
+def get_index_settings(user_settings):
+        if scraper_runner.args:
+            loop_mode = not scraper_runner.shutting_down
+            media_mode = scraper_runner.args['media_mode']
+            ids_source = scraper_runner.args['ids_source']
+        else:
+            loop_mode = False
+            media_mode = "all"
+            ids_source = "all"
+
+        status = {
+            "log_lines": get_log_file_list(),
+            "logged_in": True if "session_id" in user_settings else False,
+            "output": scraper_runner.getOutput(),
+            "scraper_settings": {
+                "loop_mode": loop_mode,
+                "media_mode": media_mode,
+                "ids_source": ids_source
+            },
+            "scraper_status": scraper_runner.getStatus()
+        }
+
+        return jsonify(status)
+
+@app.route("/api/index/", methods=['POST', 'GET'])
+def index_api():
```

Comment:
As suggested above branch it into two pieces.
Don't call it index, index it's the homepage of the website, but the BE api should not care about the frontend structure.

I'd call it scraper_settings and scraper_status